site/blog/2022-10-28-optimize-s3-endpoints.mdx

```diff
@@ -0,0 +1,126 @@
+essential for security services or when hosting any content with sensitive data.
+
+**Security: policies**
+
+Many of the servcies are supporting the use of policies, that can be used to determine the traffic that can
+pass through the interface.
+
+**Performance: latency**
+
+Because of the traffic staying within Amazon’s networks, the physical distance travelled, the number of hops
+and the risk of traversing congested networks are all significantly lower.
+
+**Performance: bandwidth**
+
+PrivateLink supports a sustained bandwidth of 10 Gbps per availability zone with bursts up to 40 Gbps.
+
+**Performance: stability**
+
+PrivateLink consistently lowers the amount of errors and timeouts on high loads
+
+**Do you want to know if you have it properly configured? Please use this query to check it:**
+
+<!--- https://www.urlencoder.org/ -->
+
+<p>
+  <a href="https://app.iasql.com/#/button/Check S3 endpoints/--%20Installing%20the%20needed%20modules%0ASELECT%20iasql_install%28%27aws_s3%27%29%3B%0ASELECT%20iasql_install%28%27aws_vpc%27%29%3B%0A--%20Sync%20the%20existing%20data%0Aselect%20%2A%20from%20iasql_sync%28%29%3B%0A--%20Perform%20the%20query%20for%20endpoints%0Aselect%20bucket.region%2C%20vpc.is_default%2C%20vpc.cidr_block%2C%0A%28select%20count%28%2A%29%3E0%20from%20endpoint_gateway%20where%20endpoint_gateway.region%3Dbucket.region%20and%20service%3D%27s3%27%20and%20endpoint_gateway.vpc_id%3Dvpc.id%29%20as%20has_endpoint_gateway%2C%0A%28select%20count%28%2A%29%3E0%20from%20endpoint_interface%20where%20endpoint_interface.region%3Dbucket.region%20and%20service%3D%27s3%27%20and%20endpoint_interface.vpc_id%3Dvpc.id%29%20as%20has_endpoint_interface%0Afrom%20bucket%20left%20outer%20join%20vpc%20on%20vpc.region%3Dbucket.region%3B">
+    <button class="button button--lg button--primary">Check your endpoints</button>
+  </a>
+</p>
+
+This query will find your active S3 buckets for all regions, associated with the existing endpoint gateways or interfaces - if they exist.
+
+Have you found missing endpoints? No problem, IaSQL can generate them for you. You can create two different
+types of endpoints: gateway and interface. We're going to describe their main features and you could create
+the desired type in an automated way.
+
+## Interface endpoints
+
+An interface endpoint is an elastic network interface with a private IP address that serves as an entry
+point for traffic destined to a supported AWS service. As they use ENIs, they can benefit of security groups
+to control traffic. Policies can also be added to control access to those endpoints.
+
+They are regional, meaning they can only be accessed by the same region they are created. Multi-region is
+possible using VPC peering, so resources in one region can be accessed from others. They only support IPv4
+TCP traffic.
+
+An interface endpoint (except S3 interface endpoint) has corresponding private DNS hostnames, that can be
```

Comment:
I think Dynamo also has the same issue? This may be a good place to link to AWS documentation on this, in any case, because "why?" is my immediate question but we start going very off-topic if we try to answer it.

Reply:
```suggestion
An interface endpoint (except S3 interface endpoint) has a corresponding private DNS hostname, that can be
```